src/riak_ql.yrl.tests

```diff
@@ -3,8 +3,13 @@
 
 -define(sql_compilation_assert(String, Expected),
 	Toks = riak_ql_lexer:get_tokens(String),
-	{ok, Got} = parse(Toks),
-	?assertEqual(Expected, Got)).
+	Got = parse(Toks),
+	?assertEqual({ok, Expected}, Got)).
+
+-define(sql_compilation_fail(QL_string),
+	Toks = riak_ql_lexer:get_tokens(QL_string),
+	Got = parse(Toks),
+	?assertMatch({error, _}, Got)).
```

Comment:
The parser may fail in many and various ways, I assume; I wonder if it would make sense to also check for a specific expected reason here?